config.go

```diff
@@ -1616,6 +1622,35 @@ func ValidateConfig(cfg Config, interceptor signal.Interceptor, fileParser,
 		cfg.DB.Bolt.NoFreelistSync = !cfg.SyncFreelist
 	}
 
+	// Parse any extra sqlite pragma options that may have been provided
+	// to determine if they override any of the defaults that we will
+	// otherwise add.
+	sqlitePragmaOpts := make(
+		map[string]bool, len(cfg.DB.Sqlite.PragmaOptions),
+	)
+	for _, option := range cfg.DB.Sqlite.PragmaOptions {
+		pair := strings.Split(option, "=")
+		if len(pair) != 2 { //nolint:gomnd
+			return nil, fmt.Errorf("invalid sqlite pragma "+
+				"option: %v. Expected the format: name=value",
+				option)
+		}
+
+		sqlitePragmaOpts[pair[0]] = true
+	}
+
+	if !sqlitePragmaOpts["synchronous"] {
+		cfg.DB.Sqlite.PragmaOptions = append(
+			cfg.DB.Sqlite.PragmaOptions, "synchronous=normal",
+		)
+	}
+
+	if !sqlitePragmaOpts["auto_vacuum"] {
+		cfg.DB.Sqlite.PragmaOptions = append(
+			cfg.DB.Sqlite.PragmaOptions, "auto_vacuum=incremental",
```

Comment:
Apparently `auto_vacuum` does not reclaim deleted rows (https://www.sqlite.org/pragma.html#pragma_auto_vacuum). What's your thoughts on this? It seems that if you close a channel and want to get the disk space back, you still need a manual `vacuum`. Should it be an option to perform this on startup, similar to bbolt auto compaction?

Reply:
yes. it is actually the `incremental_vacuum` flag that must be set (not `vacuum` as that only needs to be run if `auto_vacuum` is added after the first connection). But since we have exposed `PragmaOptions` to the users, they already have the ability to set this. 

However, I forgot to add this to the docs, so I will add a note about it there 👍

Reply:
ah - also just realised not all pragma options are kv pairs. So need to update the config to allow for non kv pairs

Reply:
updated the docs and removed the "kv" form requirement 👍

Reply:
I tried to get vacuum to clean up a test db, but didn't succeed.

This is what I did:
```
~ sqlite3 /tmp/test.db
SQLite version 3.37.0 2021-12-09 01:34:53
Enter ".help" for usage hints.
sqlite> pragma auto_vacuum=incremental;
sqlite> pragma incremental_vacuum;
sqlite> create table x (key blob primary key);
sqlite> insert into x (key) values (readfile('Downloads/vlc-3.0.17.3-arm64.dmg'));
sqlite> delete from x;
```

The db file is still big at this point. Only issuing `vacuum;` brings it back to a tiny size.

Am I missing something?

Reply:
Ah right, after all this, I need to do `pragma incremental_vacuum` again and that cleans it up. All clear :+1:

Reply:
Too bad there is no real auto-vacuum in sqlite.

Reply:
> Ah right, after all this, I need to do pragma incremental_vacuum again and that cleans it up. All clear

Cool - yeah the `pragma incremental_vacuum` only does something once there is something to vacuum. It is not a set-and-forget pragma